File: JavaRMI/src/cliente/MenuCliente.java
package cliente;

import java.util.*;
import common.*;

// Classe para o usuario navegar pelo menu
public class MenuCliente {

    //cliente emite ordem de compra ou venda (usuario, codigo acao, quantidade, valor, prazo)

    private StringBuilder menuString;
    private static Scanner keyboard;
    private CliImpl cliente;

    private String nomeDeUsuario;

    public MenuCliente(CliImpl cliente) {

        this.cliente = cliente;
        nomeDeUsuario = cliente.getNomeDeUsuario();
      
        buildMenuString();
        
        keyboard = new Scanner(System.in);
                
    }

    

    public void start() throws Exception {

        System.out.println("Welcomed " + nomeDeUsuario + "!");
        
        

        while(true){                    
            
            System.out.println();
            System.out.println("Digite o numero da opcao desejada: ");
            System.out.println(menuString);

            System.out.print(nomeDeUsuario + ": ");
            //keyboard = new Scanner(System.in);
            String option = keyboard.nextLine();
            
            if(option.isEmpty()){
                System.out.println("Opcao nao pode ser vazia");
                continue;
            }


            switch (option) {
                
                // 1) Emitir ordem de compra ou venda;
                case "1":
                    ConstrutorDeOrdem construtorDeOrdem = new ConstrutorDeOrdem(nomeDeUsuario, cliente, keyboard);
                    Ordem ordem = construtorDeOrdem.obterOrdemDoUsuario();
                    cliente.emitirOrdemDeCompraOuVenda(ordem);
                    break;
                
                // 2) Visualizar minha Lista de Interesse/Cotacoes;
                case "2":
                    cliente.obterCotacoesDaListaDeInteresse();
                    break;
                
                // 3) Atualizar minha Lista de Interesse/Cotacoes;
                case "3":
                    ConstrutorDeInteresse construtorDeInteresse = new ConstrutorDeInteresse(nomeDeUsuario, cliente, keyboard);
                    Interesse interesse = construtorDeInteresse.obterInteresseDoUsuario();
                    cliente.atualizarListaDeInteresse(interesse);
                    break;
                
                // 4) Visualizar minha carteira de acoes;   
                case "4":
                    cliente.obterCarteira();                    
                    break;
                
                // 5) Visualizar minha lista de limite de ganho/perca;    
                case "5":
                    cliente.obterListaDeLimite();
                    break;
                
                // 6) Atualizar minha lista de limite de ganho/perca;
                case "6":
                    ConstrutorDeLimite construtorDeLimite = new ConstrutorDeLimite(nomeDeUsuario, cliente, keyboard);
                    Limite limite = construtorDeLimite.obterLimiteDoUsuario();
                    cliente.atualizarListaDeLimite(limite);
                    break;
            
                // 7) Sair;
                case "0":
                    cliente.sair();
                    keyboard.close();                    
                    System.exit(0);
                    break;                    

                default:
                    System.out.println("Opcao invalida");                    
                    break;
            } 

        }
        
    }

    private void buildMenuString() {
        menuString = new StringBuilder();

        menuString.append("1) Emitir ordem de compra ou venda;\n");
        menuString.append("2) Visualizar minha lista de interesse de cotacao;\n");
        menuString.append("3) Atualizar minha lista de interesse de cotacao;\n");
        menuString.append("4) Visualizar minha carteira de acoes;\n");
        menuString.append("5) Visualizar minha lista de limite de ganho/perca;\n");
        menuString.append("6) Atualizar minha lista de limite de ganho/perca;\n");
        menuString.append("0) Sair;");
    }

}

File: Web Services/src/cliente/menu_cliente.py
import sys

from construtor_de_ordem import ConstrutorDeOrdem
from construtor_de_interesse import ConstrutorDeInteresse
from construtor_de_limite import ConstrutorDeLimite
from loggable import Loggable
from models.ordem import Ordem
from models.interesse import Interesse
from cliente_http import ClienteHttp
import os


class MenuCliente(Loggable):
    """Class responsible for getting the user inputs"""

    def __init__(self, username, cliente_http):
        super().__init__()
        self.nome_de_usuario = username
        self.cliente_http: ClienteHttp = cliente_http

    def start_menu(self):
        """ Prints the menu to the user and get keyboard input """

        self.menu: dict = {}
        self.menu['1'] = "1) Emitir ordem de compra ou venda;" 
        self.menu['2'] = "2) Visualizar minha lista de interesse de cotacao;"
        self.menu['3'] = "3) Atualizar minha lista de interesse de cotacao;"
        self.menu['4'] = "4) Visualizar minha carteira de acoes;"
        self.menu['5'] = "5) Visualizar minha lista de limite de ganho/perca;"
        self.menu['6'] = "6) Atualizar minha lista de limite de ganho/perca;"
        self.menu['7'] = "0) Sair;"

        print("Welcomed {}".format(self.nome_de_usuario))

        while True: 
            print()
            options=sorted(self.menu.keys())
            for entry in options: 
                print (self.menu[entry])

            selection=input("Digite o numero da opcao desejada: ") 
            
            if selection =='1':                 
                print (self.menu[selection])
                try:
                    construtor_de_ordem = ConstrutorDeOrdem(self.nome_de_usuario)
                    ordem: Ordem = construtor_de_ordem.obter_ordem_do_usuario()
                    self.cliente_http.emitir_ordem(ordem)
                except:
                    self.log_error("Nao foi possivel emitir a ordem", exc=sys.exc_info()[0])
                
            elif selection == '2': 
                print (self.menu[selection])
                try:
                    self.cliente_http.obter_interesses()
                except:
                    self.log_error("Nao foi possivel obter cotacoes da lista de interesse", exc=sys.exc_info()[0])
            
            elif selection == '3': 
                print (self.menu[selection])
                try:
                    construtor_de_interesse = ConstrutorDeInteresse(self.nome_de_usuario)
                    interesse = construtor_de_interesse.obter_interesse_do_usuario()
                    self.cliente_http.enviar_interesse(interesse)
                except:
                    self.log_error("Nao foi possivel enviar novo interesse", exc=sys.exc_info()[0])

            elif selection == '4': 
                print (self.menu[selection])
                try:
                    self.cliente_http.obter_carteira()
                except:
                    self.log_error("Nao foi possivel obter carteira", exc=sys.exc_info()[0])
                    
            elif selection == '5': 
                print (self.menu[selection])
                try:
                    self.cliente_http.obter_limites()
                except:
                    self.log_error("Nao foi possivel obter limites", exc=sys.exc_info()[0])

            elif selection == '6': 
                print (self.menu[selection])
                try:
                    construtor_de_limite = ConstrutorDeLimite(self.nome_de_usuario)
                    limite = construtor_de_limite.obter_limite_do_usuario()
                    self.cliente_http.enviar_limite(limite)
                except:
                    self.log_error("Nao foi possivel enviar novo limite", exc=sys.exc_info()[0])                                            

            elif selection == '0': 
                os._exit(1)
                break
            else: 
                print ("Opcao invalida")
